tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7771,8 +7771,12 @@ def err_hlsl_unsupported_for_version_lower : Error<
    "%0 is only allowed for HLSL %1 and lower.">;
 def err_hlsl_unsupported_keyword_for_min_precision : Error<
    "%0 is only supported with -enable-16bit-types option">;
-def err_hlsl_intrinsic_overload_in_wrong_shader_model : Error<
-   "overload of intrinsic %0 requires shader model %1 or greater">;
+def err_hlsl_derivatives_intrinsic_in_wrong_shader_kind : Warning<
+   "Derivatives intrinsic %0 only works in lib/ps/cs/as/ms">,
```

Comment:
With the more subtle check combining stage and version, the message either needs revision, or we need different messages depending on whether we failed due to stage or shader model.

We also need to add node, and I think including 'lib' here might just add confusion, since it's not clear what this means to the user.

Reply:
Updated.